tools/Google.Cloud.Tools.ReleaseManager/UpdateFromBazelCommand.cs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Cloud.Tools.Common;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Google.Cloud.Tools.ReleaseManager;
+
+/// <summary>
+/// A class to update parameters in the API catalog based on BUILD.bazel files.
+/// (This is a stopgap measure until we've got a single configuration source.)
+/// </summary>
+public sealed class UpdateFromBazelCommand : CommandBase
+{
+    public UpdateFromBazelCommand() : base("update-from-bazel", "Updates the API catalog based on BUILD.bazel files", "id")
+    {
+    }
+
+    protected override void ExecuteImpl(string[] args)
+    {
+        string id = args[0];
+        var catalog = ApiCatalog.Load();
+
+        var ids = id == "all"
+            ? catalog.Apis.Where(api => api.ProtoPath is string && api.Generator == GeneratorType.Micro).Select(api => api.Id).ToArray()
+            : new[] { id };
+
+        bool anyUpdated = false;
+        foreach (var apiId in ids)
+        {
+            if (Update(catalog[apiId]))
+            {
+                anyUpdated = true;
+                Console.WriteLine($"Modified configuration for {apiId}");
+            }
+        }
+
+        if (anyUpdated)
+        {
+            File.WriteAllText(ApiCatalog.CatalogPath, catalog.FormatJson());
+        }
+    }
+
+    public bool Update(ApiMetadata api)
+    {
+        var root = DirectoryLayout.DetermineRootDirectory();
+        var googleapis = Path.Combine(root, "googleapis");
+
+        var bazelFile = Path.Combine(googleapis, api.ProtoPath, "BUILD.bazel");
+        if (!File.Exists(bazelFile))
+        {
+            throw new UserErrorException($"Expected Bazel file does not exist: '{bazelFile}'");
+        }
+
+        var lines = File.ReadAllLines(bazelFile).ToList();
+        var start = lines.IndexOf("csharp_gapic_library(");
+        if (start == -1)
+        {
+            // This is effectively a warning. (AI Platform has it commented out, for example.)
```

Comment:
Wouldn't you still find it if it's commented out?

Reply:
Nope - because then the line is `# csharp_gapic_library`.

Reply:
Ah yes, you are looking for whole lines, not within a string. Duh.